src/tools/pen.rs

```diff
@@ -88,6 +114,26 @@ impl Tool for Pen {
         self.this_edit_type.take()
     }
 
+    fn key_down(
+        &mut self,
+        event: &KeyEvent,
+        _ctx: &mut EventCtx,
+        data: &mut EditSession,
+        _: &Env,
+    ) -> Option<EditType> {
+        assert!(self.this_edit_type.is_none());
+        match event {
+            e if e.key == KbKey::Backspace => {
+                data.delete_selection();
+                self.this_edit_type = Some(EditType::Normal);
+            }
+            // TODO: should support nudging; basically a lot of this should
```

Comment:
this is something I've been thinking about; a lot of the details of selection manipulation should always be available. Maybe that should be implemented as some kind of shared component.